hawt-app-maven-plugin/src/main/resources/org/jboss/hawt/app/maven/bin/run

```diff
@@ -1,53 +1,46 @@
 #!/bin/sh
 #
-# Discover the APP_HOME from the location of this script.
+# Discover the JAVA_APP_DIR from the location of this script.
 #
-if [ -z "${APP_HOME}" ] ; then
+if [ x"${JAVA_APP_DIR}" == x ] ; then
   DIRNAME=`dirname "$0"`
-  APP_HOME=`cd "$DIRNAME"/.. ; pwd`
-  export APP_HOME
+  JAVA_APP_DIR=`cd "$DIRNAME"/.. ; pwd`
+  export JAVA_APP_DIR
+fi
+
+if [ x"$JAVA_APP_USER"  != x ] && [ `id -un` != "$JAVA_APP_USER" ] ; then
+    # re-run the launch under the right user id...
+    exec sudo -n -u ${JAVA_APP_USER} $0 $@
```

Comment:
Just copied that over from the original script. But you are right, should be `su`. will fix that.